### PR TITLE
Feature/rate limit options

### PR DESCRIPTION
**Pull request by bcksl**
## Description

- Update `express-rate-limit` to current version;
- Derive `config.rateLimit` type from `express-rate-limit`'s `Options`;
- `ts-essentials` updated;
- `@types/express-rate-limit` removed (deprecated, types now in main package).

- [x] I have read and understand the CONTRIBUTING.md document in this repository

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Existing test suite passes locally with my changes
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/express/middleware/index.ts [hunk 1]
```diff
@@ -6,26 +6,18 @@ import methodOverride from 'method-override';
...
-    windowMs?: number
-    max?: number
-    skip?: (req: PayloadRequest) => boolean
-  } = {
-    windowMs: payload.config.rateLimit.window,
-    max: payload.config.rateLimit.max,
-  };
-
-  if (typeof payload.config.rateLimit.skip === 'function') rateLimitOptions.skip = payload.config.rateLimit.skip;
+  const rateLimitOptions = _.omit(payload.config.rateLimit, 'window', 'trustProxy');
```
Comment (DanRibbens) [inline src/express/middleware/index.ts:19]: I don't think we want to pull in all of `lodash` for one function here. Can you update this to work without _.omit()?

**Review by DanRibbens — CHANGES_REQUESTED** [review]
Thanks for the PR, making one change request. 👍

File: src/express/middleware/index.ts (hunk 1 above)
Comment (evelynhathaway) [reply src/express/middleware/index.ts:19]: I ran across this MR and wanted to add a suggestion that doesn't require Lodash or a util function

```suggestion
  const {window, trustProxy, ...rateLimitOptions} = payload.config.rateLimit;
```

**Comment by denolfe** [comment]
@bcksl Revisiting this one. I think implementing what was suggested above and removing the lodash dependency would make most sense. 

While it is a dependency of some of our dependencies, we prefer to avoid using lodash explicitly in our codebase (for now).

https://github.com/payloadcms/payload/pull/2012#discussion_r1162146159

**Comment by denolfe** [comment]
Closing for now. Feel free to reopen if you update it 👍

**Pull request closed without merging**